_rules/SC3-1-2-lang.md

```diff
@@ -1,79 +1,78 @@
-- This test checks the `lang` attributes. The `xml:lang` attributes are not taken into account because tests have shown, that `xml:lang` is ignored by screenreaders. (Both Jaws 15 with FF and IE and NVDA with FF go by lang attribute, xml:lang is ignored.) The `xml:lang` attributes are checked by a separate test: [SC3-1-2-xml-lang](SC3-1-2-xml-lang.html).
+**Note**: To retrieve all DOM nodes with a `lang` and or `xml:lang` attribute, other than the root node use -
+`descendent-or-self::body[@lang]` -or-  `descendent-or-self::body[@xml:lang]`.
 
-## Test procedure
+### Expectation
 
-### Selector
+- The value of the `lang` and/or `xml:lang` conforms to [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt).
 
-Select all elements that match the following XPATH selector:
+## Assumptions
 
-    descendent-or-self::body[@lang]
+*There are currently no assumptions*
 
-This test is applied to all elements with `lang` attribute in the body of the web page (including the body element itself).
+## Accessibility support
 
-For each selected item, go through the following steps:
+Most assistive technologies will rely solely on the `lang` attribute for determining the language of the page. The requirement tested in this rule will only give rise to accessibility issues in cases where assistive technologies are inconsistent about how they determine the language of the page.
 
-### Step 1
+## Background
 
-L1 = value of `lang` attribute.
+- https://www.w3.org/TR/2014/NOTE-WCAG20-TECHS-20140408/H58
+- https://www.ietf.org/rfc/bcp/bcp47.txt
+- http://wiki.egovmon.no/wiki/SC3.1.2#Element_descendent-or-self::body.5B.40lang.5D_or_descendent-or-self::body.5B.40xml:lang.5D
+- https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=312#qr-meaning-other-lang-id
 
-Compare L1 to BCP 47.
+## Test Cases
 
-If L1 is not on the list, return [step1-fail](#step1-fail)
+### Passed
 
-Else, return [step1-pass](#step1-pass)
+```html
+<article lang="en"></article>
+```
 
-*Note that this step also fails if L1 contains only whitespace or is empty.*
+```html
+<p xml:lang="DE"></p>
+```
 
-## Outcome
+```html
+<blockquote lang="fr-CH"></blockquote>
+```
 
-The resulting assertion is as follows,
+### Failed
 
-| Property | Value
-|----------|----------
-| type     | Assertion
-| test     | auto-wcag:{{ page.rule_id }}
-| subject  | *the selected element*
-| mode     | auto-wcag:{{ page.test_mode }}
-| result   | <One TestResult from below>
+```html
+<article lang="dutch"></article>
+```
 
-### step1-fail
+```html
+<p xml:lang="english"></p>
+```
 
-| Property    | Value
-|-------------|----------
-| type        | TestResult
-| outcome     | Failed
-| description | Unknown language code L1
+### Inapplicable
 
-### step1-pass
+```html
+<article lang=""></article>
+```
 
-| Property    | Value
-|-------------|----------
-| type        | TestResult
-| outcome     | Passed
-| description |
+[non-empty]: ../pages/algorithms/non-empty.html
```

Comment:
Fixing the above ambiguity and adding test cases where both `lang` and `xml:lang` are specified would be great.